components/more_direct_channels/group_message_option.jsx

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {localizeMessage} from 'utils/utils';
+
+const GroupMessageOption = (props) => {
+    const {channel, isSelected, onAdd} = props;
+
+    let className = 'mentions__name';
+    if (isSelected) {
+        className += ' suggestion--selected';
+    }
+
+    const displayName = channel.profiles.map((profile) => '@' + profile.username).join(', ');
+    const icon = <div className='more-modal__gm-icon bg-text-200'>{channel.profiles.length}</div>;
+
+    const addValue = () => {
```

Comment:
Could you make this into a component and make this a proper method? Using an arrow function when rendering a component is discouraged